Source/processcontainers/implementations/CRunImplementation/CRunImplementation.cpp

```diff
@@ -0,0 +1,460 @@
+                if (ret != 0) {
+                    TRACE_L1("Failed to run a container \"%s\". Error %d: %s", _name.c_str(), error->status, error->msg);
+                    result = false;
+                } else {
+                    _created = true;
+                }
+            }
+        }
+
+        return result;
+    }
+
+    bool CRunContainer::Stop(const uint32_t timeout /*ms*/)
+    {
+        bool result = true;
+        libcrun_error_t error = nullptr;
+
+        if (libcrun_container_delete (&_context, NULL, _name.c_str(), true, &error) != 0) {
+            TRACE_L1("Failed to stop a container \"%s\". Error: %s", _name.c_str(), error->msg);
+            result = false;
+        }
+        _created = false;
+
+        return result;
+    }
+
+    void CRunContainer::AddRef() const 
+    {
+        _refCount++;
+    };
+
+    uint32_t CRunContainer::Release()
+    {
+        if (--_refCount == 0) {
+            if (_created == true) {
+                Stop(Core::infinite);
+            }        
+        }
+
+        return Core::ERROR_NONE;
+    };
+
+    void CRunContainer::OverwriteContainerArgs(libcrun_container_t* container, const string& newComand, IStringIterator& newParameters)
+    {
+        // Clear args that were set by runtime
+        if (container->container_def->process->args) {
+            for (size_t i = 0; i < container->container_def->process->args_len; i++)
+            {
+                if (container->container_def->process->args[i] != NULL)
+                {
+                    free (container->container_def->process->args[i]);
+                }
+            }
+            free (container->container_def->process->args);
+        }
+
+        char** argv = reinterpret_cast<char**>(malloc((newParameters.Count() + 2) * sizeof(char*)));
+        int argc = 0;
+
+        // Set custom starting command
+        argv[argc] = reinterpret_cast<char*>(malloc(sizeof(char) * newComand.length() + 1));
+        strcpy(argv[argc], newComand.c_str());
+        argc++;
+
+        // Set command arguments
+        while (newParameters.Next())
+        {
+            argv[argc] = reinterpret_cast<char*>(malloc(sizeof(char) * newParameters.Current().length() + 1));
+            strcpy(argv[argc], newParameters.Current().c_str());
+            argc++;
+        }
+        argv[argc] = nullptr;
+        container->container_def->process->args = argv;
+        container->container_def->process->args_len = argc;
+    }
+
+    uint32_t CRunContainer::ClearLeftovers() 
+    {
+        libcrun_error_t error = nullptr;
+        int ret = 0;
+
+        // Find containers 
+        libcrun_container_list_t *list;
+        ret = libcrun_get_containers_list (&list, "/run/crun", &error);
+        if (ret < 0) {
+            TRACE_L1("Failed to get containers list. Error %d: %s", error->status, error->msg);
+            return Core::ERROR_UNAVAILABLE;
```

Comment:
Minor, but still for the next time, for maintainability reasons, the coding guidelines only allow for 1 return statement in a method.